// node-scp/lib/constant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CLIENT_EVENTS = exports.targetType = exports.errorCode = void 0;
var errorCode;
(function (errorCode) {
    errorCode["generic"] = "ERR_GENERIC_CLIENT";
    errorCode["connect"] = "ERR_NOT_CONNECTED";
    errorCode["badPath"] = "ERR_BAD_PATH";
    errorCode["permission"] = "EACCES";
    errorCode["notexist"] = "ENOENT";
    errorCode["notdir"] = "ENOTDIR";
})(errorCode = exports.errorCode || (exports.errorCode = {}));
var targetType;
(function (targetType) {
    targetType[targetType["writeFile"] = 1] = "writeFile";
    targetType[targetType["readFile"] = 2] = "readFile";
    targetType[targetType["writeDir"] = 3] = "writeDir";
    targetType[targetType["readDir"] = 4] = "readDir";
    targetType[targetType["readObj"] = 5] = "readObj";
    targetType[targetType["writeObj"] = 6] = "writeObj";
})(targetType = exports.targetType || (exports.targetType = {}));
exports.CLIENT_EVENTS = new Set([
    "banner",
    "ready",
    "tcp connection",
    "x11",
    "keyboard-interactive",
    "change password",
    "error",
    "end",
    "close",
    "timeout",
    "connect",
    "greeting",
    "handshake",
    "hostkeys",
    "unix connection",
]);
